packages/generator-emakinacee-react/generators/component/index.js

```diff
@@ -34,39 +35,60 @@ class AppGenerator extends Generator {
             templatesToCopy.push('component-test.txt');
         }
 
-        this._copyFiles(templatesToCopy);
+        this._copyFiles(templatesToCopy, pathOptions);
+    }
+
+    _getPathOptions() {
+        const pathOptions = {};
+        const split = this.options.path.split('/');
+
+        if (split[0] === '') split.shift();
+        if (split[0] === '.') {
+            pathOptions.noDefaultFolder = true;
+            split.shift();
+        }
+        pathOptions.componentName = this._getName(split.pop());
+        pathOptions.path = split.join('/');
+
+        return pathOptions;
```

Comment:
Could be replaced with this, except that `_getName` is not called:
`path.replace(/^\//, '').match(/^(?<noDefault>\.\/)?(?<path>.*)\/(?<componentName>[^\/]+)$/).groups`

Reply:
what a nice regex XD